mobilecoind/src/conversions.rs

```diff
@@ -315,6 +365,84 @@ mod test {
         assert_eq!(rust, UnspentTxOut::try_from(&proto).unwrap());
     }
 
+    // Test the From<&MemoPayload> for api::DecodedMemo implementation
+    #[test]
+    fn test_memo_conversion() {
+        let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+
+        let alice = AccountKey::new(
+            &RistrettoPrivate::from_random(&mut rng),
+            &RistrettoPrivate::from_random(&mut rng),
+        );
+        let alice_cred = SenderMemoCredential::from(&alice);
+        let alice_hash = alice_cred.address_hash.clone();
+
+        let bob = AccountKey::new(
+            &RistrettoPrivate::from_random(&mut rng),
+            &RistrettoPrivate::from_random(&mut rng),
+        );
+        let bob_addr = bob.default_subaddress();
+
+        let tx_public_key = CompressedRistrettoPublic::from_random(&mut rng);
+
+        let memo1 = UnusedMemo {};
+        let decoded = decode_memo(&MemoPayload::from(memo1));
+        assert!(!decoded.has_authenticated_sender_memo());
+        assert!(!decoded.has_unknown_memo());
```

Comment:
similar to below this won't scale as more items are added to a one of perhaps
```suggestion
        assert_eq!(decoded.decoded_memo, None);
```